noteable/datasources.py

```diff
@@ -86,42 +99,53 @@ def bootstrap_datasource(
-    human_name = metadata.get('name')
-
-    # Do any per-drivername post-processing of and dsn_dict and create_engine_kwargs
-    # before we make use of any of their contents. Post-processors may end up rejecting this
-    # configuration, so catch and handle just like a failure when calling Connection.set().
-    if drivername in post_processor_by_drivername:
-        post_processor: Callable[[str, dict, dict], None] = post_processor_by_drivername[drivername]
-        try:
+    human_name = metadata.get('name', 'Unnamed legacy connection')
+
+    sql_cell_handle = f'@{datasource_id}'
+
+    try:
+        # Do any per-drivername post-processing of and dsn_dict and create_engine_kwargs
+        # before we make use of any of their contents. Post-processors may end up rejecting this
+        # configuration, so catch and handle just like a failure when calling Connection.set().
+        if drivername in post_processor_by_drivername:
+            post_processor: Callable[[str, dict, dict], None] = post_processor_by_drivername[
+                drivername
+            ]
             post_processor(datasource_id, dsn_dict, create_engine_kwargs)
-        except Exception as e:
-            logger.exception(
-                'Unable to bootstrap datasource',
-                datasource_id=datasource_id,
-                datasource_name=human_name,
-            )
-
-            # Remember the failure so can be shown if / when human tries to use the connection.
-            noteable.sql.connection.Connection.add_bootstrapping_failure(
-                datasource_id, human_name, str(e)
-            )
-
-            # And bail early.
-            return
-
-    # Ensure the required driver packages are installed already, or, if allowed,
-    # install them on the fly.
-    ensure_requirements(
-        datasource_id,
-        metadata['required_python_modules'],
-        metadata['allow_datasource_dialect_autoinstall'],
-    )
+
+        # Ensure the required driver packages are installed already, or, if allowed,
+        # install them on the fly.
+        ensure_requirements(
+            datasource_id,
+            metadata['required_python_modules'],
+            metadata['allow_datasource_dialect_autoinstall'],
+        )
+
+    except Exception as e:
+        # Exception from either post_processor() or ensure_requirements().
+        logger.exception(
+            'Unable to bootstrap datasource',
+            datasource_id=datasource_id,
+            datasource_name=human_name,
+        )
+
+        # Remember the failure so can be shown if / when human tries to use the connection.
+        connection_registry.add_bootstrapping_failure(sql_cell_handle, human_name, str(e))
+
+        # And bail early.
+        return
 
     # Prepare connection URL string.
     url_obj = URL.create(**dsn_dict)
     connection_url = str(url_obj)
 
+    # XXX TODO, make a mixin for future SQLAlchemy DisableAutoCommit subclasses incorporating
```

Comment:
I don't need a ticket for this, got whole development arc for Great Success in my heaad. Will write up google doc proposal / project plan to prove it, though.